### ISSUE TITLE
Outline process for setting up development environment on Pantheon

### ISSUE THREAD

**Issue by gusaus** [DrupalOpenLearning/drupaldojo#2]
We should document how we the development environment on and code repo are set up. Did we fork this repo? https://github.com/lquessenberry/OpenSocialOnPantheon

Are we able to pull in the latest Open Social code this way? What else do we need set up on Pantheon? I believe we have the ability to set up Multidev if that's what we need.
https://pantheon.io/docs/pantheon-workflow/

Do we need someone to help manage the Pantheon environment?

**Comment by johnatasjmo** [comment]
code has been force pushed to pantheon using instructions on https://github.com/lquessenberry/OpenSocialOnPantheon

This is a composer flavour site, therefore all module updates should be done using composer.

For better deployment, we should create one environment per pull request using multidev in pantheon.  Someone else should take the lead for development now that site is set up. Pantheon has very good documentation.

**Comment by ikit-claw** [comment]
Okay, I will get onto documenting this! Thanks

**Comment by gusaus** [comment]
@ikit-claw I don't believe the code has been updated since @johnatasjmo set up per https://github.com/DrupalOpenLearning/drupaldojo/issues/2#issuecomment-313266103

Possibly you could follow up here or in Slack regarding what you need to document and update the code per the instructions? 

If it turns out we need something else from Pantheon, we know some people on the inside. DrupalOpenLearning/drupaldojo-old#1 

I'll set you up as admin.

**Comment by gusaus** [comment]
Think we can add @ChandeepKhosa to the help with this task if needed queue:

> happy to assist with development with open social on pantheon, very familiar with the platform. got some :drupal8: experience too, although a sitebuilder & themer

@miesz-ko posted something related in #opensocial -

> we're aware that it probably should be easier to run Open Social on other (Drupal) cloud platforms, but for now we consciously decide to get the security advisory done first.
any help is appreciated, more info here : https://www.drupal.org/node/2898988

We should be able to direct new contributors to Open Social once we get this site and Dojo program rolling again.

**Comment by ikit-claw** [comment]
I will follow up with Chan this week and see if he wants to have a pop at updating the site.

**Comment by ikit-claw** [comment]
Everything I am doing is being documented Will close this ticket.

**Comment by gusaus** [comment]
@ikit-claw I think we should have our dev process posted on the site where those working on the site can reference. Possibly we can first post in our wiki?

**Comment by gusaus** [comment]
In addition to outlining our Pantheon dev process, we still need someone to take on the Pantheon/sysadmin role. Per https://github.com/elmsln/openulmus.org/issues/3#issuecomment-321382520 this could be a potential role for @danielrose28 ?

**Comment by gusaus** [comment]
@ikit-claw Per our Slack convo, I think having the process documented will make it easier for contributors. I don't see anything wrong with making Dojo flavored modifications to the existing Pantheon docs - 

- https://pantheon.io/docs/pantheon-workflow/
- https://pantheon.io/docs/drupal-8-configuration-management/

We also want to make sure we're following along with our current development environment.

**Comment by gusaus** [comment]
@cleverington Per our discussion regarding your recommendations for incorporating local dev tools into our Pantheon environment https://github.com/drupaldiversity/contrib/issues/17#issuecomment-339180777 

We have a sponsored Pantheon professional environment, which I think could provide a wide range of learning opportunities.

If you’re interested helping manage our workflows, and at the same time setting up one potential dev environment other learning groups could use, possibly you could help @ikit-claw on this task?

**Comment by cleverington** [comment]
## Small Update

Spoke with the friend who works with Kalabox and Lando a lot.

The GUI for Lando is 'planned', but probably will be sometime in another year, so it would probably be okay to move forward with using Kalabox.io -based instructions for those who want to be contributors.

**Comment by gusaus** [comment]
@cleverington I think I can speak for everybody who has set up and managed the environment so far. Whatever you recommend and are comfortable with is great!  Kalabox and Lando being open source projects https://www.kalabox.io/ may also provide some opportunity to get contributors involved with those projects via Drupal Dojo sessions and mentoring projects.

So are you interested in taking the lead on this? As the building and management of the Dojo site is itself an ongoing learning project, we're hoping to find an apprentice who can gain experience by helping.

**Comment by gusaus** [comment]
@cleverington Would a new release of https://www.drupal.org/project/social and https://github.com/DrupalOpenLearning/drupaldojo/issues/61 be a good reason to take a look at our Pantheon development environment? 

Surely @ikit-claw will happily walk your through the current setup/Open Social workflow if it means he doesn't have to feel the pain of the last update!

**Comment by gusaus** [comment]
Based on our convo in Slack, I 'think' we've decided to create a custom upstream for the Dojo and subsequent sites with similar requirements. We also need to update to the latest Open Social and add an additional module https://github.com/DrupalOpenLearning/drupaldojo/issues/61 as noted above. Also should we incorporate multidev into our workflows? Might that be helpful if additional contributors come aboard?

@cleverington Do you have access to everything you need to set this up? Not sure if we need to create a new repo or a branch here? https://github.com/DrupalOpenLearning/drupaldojo

**Comment by gusaus** [comment]
@dsquaredb Given the fact we'll be in a better position to document our process after we refine our ddev flow, after we're set up... aside from adding you to our Pantheon team, what would you suggest in terms of updating our code? 

Assuming we should wait for the impending Open Social security release https://www.drupal.org/project/social/issues/2956045 before we determine what (if any) issues may come up with an upgrade from https://www.drupal.org/project/social/releases/8.x-1.5?

**Comment by gusaus** [comment]
@dsquaredb I think most of Drupal Slack went quiet for a good bit of time yesterday! Even I had to force myself back into developer mode.

I added your email associated with github to the Pantheon team. Ping me in Slack if you're not able to access upstreams, multidev, or anything related to the Pantheon account. 

We're wide open to suggestions on how to setup/manage the environment and additional collaborators and resources you may need. 

Thanks!

**Comment by ikit-claw** [comment]
@dsquaredb thanks for stepping up

**Comment by gusaus** [comment]
Based on further discussions with @rickmanelius we're sold on the idea of incorporating DDEV into our workflow:

> DDEV can very much be a means of maintaining upstream distros because it makes it easy to pull/stand-up and verify patches/updates.

As touched on in this issue, we're currently 'not' pulling from https://github.com/lquessenberry/OpenSocialOnPantheon because it's outdated.  

Based on that, I’m thinking we should take a fresh look at [my comment here](https://github.com/lquessenberry/OpenSocialOnPantheon/issues/2#issuecomment-363284502) and determine the scope of creating/maintaining an upstream for Open Social. 

Once that's been sorted out we can circle back to leveraging the upstream and DDEV into our Patheon workflow for the Dojo site.
